diff --git a/src/drivers/flydigi_vader_4_pro/driver.rs b/src/drivers/flydigi_vader_4_pro/driver.rs
index 91d1752..8072aa2 100644
--- a/src/drivers/flydigi_vader_4_pro/driver.rs
+++ b/src/drivers/flydigi_vader_4_pro/driver.rs
@@ -48,16 +48,23 @@ pub struct Driver {
 
 impl Driver {
     pub fn new(udevice: UdevDevice) -> Result<Self, Box<dyn Error + Send + Sync>> {
+        log::debug!("before hidrawpath");
         let hidrawpath = udevice.devnode();
+        log::debug!("before cs_path");
         let cs_path = CString::new(hidrawpath.clone())?;
+        log::debug!("before api");
         let api = hidapi::HidApi::new()?;
+        log::debug!("before open_path");
         let device = api.open_path(&cs_path)?;
 
+        log::debug!("before get_device_info");
         let info = device.get_device_info()?;
+        log::debug!("before if");
         if info.vendor_id() != VID || info.product_id() != PID {
             return Err(format!("Device '{hidrawpath}' is not a Flydigi Vader 4 Pro").into());
         }
 
+        log::debug!("before return");
         Ok(Self {
             device,
             state: None,

