diff --git a/xpad.c b/xpad.c
index 43985dd..55ac053 100644
--- a/xpad.c
+++ b/xpad.c
@@ -375,7 +375,7 @@ static const struct xpad_device {
 	{ 0x24c6, 0xfafe, "Rock Candy Gamepad for Xbox 360", 0, XTYPE_XBOX360 },
 	{ 0x2563, 0x058d, "OneXPlayer Gamepad", 0, XTYPE_XBOX360 },
 	{ 0x2dc8, 0x2000, "8BitDo Pro 2 Wired Controller for Xbox", MAP_SELECT_BUTTON, XTYPE_XBOXONE },
-	{ 0x2dc8, 0x3106, "8BitDo Ultimate Wireless / Pro 2 Wired Controller", 0, XTYPE_XBOX360 },
+	{ 0x2dc8, 0x3106, "8BitDo Ultimate Wireless / Pro 2 Wired Controller", MAP_PADDLES, XTYPE_XBOX360 },
 	{ 0x2dc8, 0x3109, "8BitDo Ultimate Wireless Bluetooth", 0, XTYPE_XBOX360 },
 	{ 0x2e95, 0x0504, "SCUF Gaming Controller", MAP_SELECT_BUTTON, XTYPE_XBOXONE },
 	{ 0x31e3, 0x1100, "Wooting One", 0, XTYPE_XBOX360 },
@@ -965,6 +965,9 @@ static void xpad360_process_packet(struct usb_xpad *xpad, struct input_dev *dev,
 		input_report_abs(dev, ABS_RZ, data[5]);
 	}
 
+	input_report_key(dev, BTN_TRIGGER_HAPPY5, data[2] & BIT(7) && data[3] & BIT(4));   // Leftmost paddle (M2)
+	input_report_key(dev, BTN_TRIGGER_HAPPY8, data[2] & BIT(7) && data[3] & BIT(5));   // Rightmost paddle (M1)
+
 	input_sync(dev);
 
 	/* XBOX360W controllers can't be turned off without driver assistance */
@@ -2242,7 +2245,7 @@ static int xpad_probe(struct usb_interface *intf, const struct usb_device_id *id
 	xpad->xtype = xpad_device[i].xtype;
 	xpad->name = xpad_device[i].name;
 	xpad->quirks = xpad_device[i].quirks;
-	xpad->packet_type = PKT_XB;
+	xpad->packet_type = PKT_XBE2_FW_5_11;
 	INIT_WORK(&xpad->work, xpad_presence_work);
 
 	if (xpad->xtype == XTYPE_UNKNOWN) {
@@ -2308,38 +2311,6 @@ static int xpad_probe(struct usb_interface *intf, const struct usb_device_id *id
 
 	usb_set_intfdata(intf, xpad);
 
-	/* Packet type detection */
-	if (le16_to_cpu(udev->descriptor.idVendor) == 0x045e) { /* Microsoft controllers */
-		if (le16_to_cpu(udev->descriptor.idProduct) == 0x02e3) {
-			/* The original elite controller always uses the oldest
-			 * type of extended packet
-			 */
-			xpad->packet_type = PKT_XBE1;
-		} else if (le16_to_cpu(udev->descriptor.idProduct) == 0x0b00) {
-			/* The elite 2 controller has seen multiple packet
-			 * revisions. These are tied to specific firmware
-			 * versions
-			 */
-			if (le16_to_cpu(udev->descriptor.bcdDevice) < 0x0500) {
-				/* This is the format that the Elite 2 used
-				 * prior to the BLE update
-				 */
-				xpad->packet_type = PKT_XBE2_FW_OLD;
-			} else if (le16_to_cpu(udev->descriptor.bcdDevice) <
-				   0x050b) {
-				/* This is the format that the Elite 2 used
-				 * prior to the update that split the packet
-				 */
-				xpad->packet_type = PKT_XBE2_FW_5_EARLY;
-			} else {
-				/* The split packet format that was introduced
-				 * in firmware v5.11
-				 */
-				xpad->packet_type = PKT_XBE2_FW_5_11;
-			}
-		}
-	}
-
 	if (xpad->xtype == XTYPE_XBOX360W) {
 		/*
 		 * Submit the int URB immediately rather than waiting for open
